firebase-firestore/src/main/java/com/google/firebase/firestore/core/SyncEngine.java

```diff
@@ -254,8 +267,29 @@ private ViewSnapshot initializeViewAndComputeSnapshot(
     return viewChange.getSnapshot();
   }
 
+  /**
+   * Sends the listen to the RemoteStore to get remote data. Invoked when a Query starts listening
+   * to the remote store, while already listening to the cache.
+   */
+  public void listenToRemoteStore(Query query) {
+    assertCallback("listenToRemoteStore");
+    hardAssert(
+        queryViewsByQuery.containsKey(query), "This is the first listen to query: %s", query);
+
+    TargetData targetData = localStore.allocateTarget(query.toTarget());
+    remoteStore.listen(targetData);
+  }
+
   /** Stops listening to a query previously listened to via listen. */
   void stopListening(Query query) {
```

Comment:
You should get rid of this method as well. I noticed there is a test `testUnlistensOnUnknownListeners` that verifies according to this method, so that will need to be changed. I suggest doing something like this:

```
verify(syncSpy, never()).stopListening(eq(query), anyBoolean());
```

Because you change the behaviour `EventManager` behaviour to use `stopListening(Query query, boolean shouldUnlistenToRemote)` instead, you actually undermine the test `verify(syncSpy, never())`.

It might be better to change verification to be:

```
verifyZeroInteractions(syncSpy);
```

That way no matter what method is called, the test will fail as intended.

Reply:
`verifyZeroInteractions(syncSpy);`  failed with this error:
 ```No interactions wanted here:
-> at com.google.firebase.firestore.core.EventManagerTest.testUnlistensOnUnknownListeners(EventManagerTest.java:85)
But found this interaction on mock 'syncEngine':
-> at com.google.firebase.firestore.core.EventManager.<init>(EventManager.java:83)
Actually, above is the only interaction with this mock.
```